packages/flutter/lib/src/material/text_field.dart

```diff
@@ -606,6 +615,18 @@ class _TextFieldState extends State<TextField> with AutomaticKeepAliveClientMixi
   @override
   bool get wantKeepAlive => _splashes != null && _splashes.isNotEmpty;
 
+  bool get _cursorOpacityAnimates => Theme.of(context).platform == TargetPlatform.iOS ? true : false;
+
+  Color get _cursorColor {
+    if (widget.cursorColor == null) {
+      if (Theme.of(context).platform == TargetPlatform.iOS)
+        return CupertinoColors.activeBlue;
```

Comment:
While you're here :D
This is actually theme dependent (see the dark themed clock app or the red themed calendar app).

In other words, this is actually `CupertinoTheme.of(context).primaryColor`.